jobmaster/dm/metadata/job.go

```diff
@@ -0,0 +1,99 @@
+package metadata
+
+import (
+	"context"
+
+	"github.com/hanfei1991/microcosm/jobmaster/dm/config"
+	"github.com/hanfei1991/microcosm/lib"
+	"github.com/hanfei1991/microcosm/pkg/adapter"
+	"github.com/hanfei1991/microcosm/pkg/meta/metaclient"
+	"github.com/pingcap/errors"
+)
+
+// TODO: use Stage in lib or move Stage to lib.
+type TaskStage int
+
+// These stages may updated in later pr.
+const (
+	StageInit TaskStage = iota
+	StageRunning
+	StagePaused
+	StageFinished
+	StageDelete
+)
+
+// Job represents the state of a job.
+type Job struct {
+	State
+
+	// taskID -> task
+	Tasks map[string]*Task
+}
+
+func NewJob(jobCfg *config.JobCfg) *Job {
+	taskCfgs := jobCfg.ToTaskConfigs()
+	job := &Job{
+		Tasks: make(map[string]*Task, len(taskCfgs)),
+	}
+
+	for taskID, taskCfg := range taskCfgs {
+		job.Tasks[taskID] = NewTask(taskCfg)
+	}
+	return job
+}
+
+// Task represents the status of a upstream task.
```

Comment:
"A job may contain multiple upstream and it will be converted into multiple tasks."